crypto/perlasm/cbc.pl

```diff
@@ -6,6 +6,8 @@
 # in the file LICENSE in the source distribution or at
 # https://www.openssl.org/source/license.html
 
+# Check if Intel CET is enabled at compile time.
+$endbr32=(`echo "#ifdef __CET__\n#error CET is enabled\n#endif" | $ENV{CC} -c -o /dev/null -x c - 2>&1` =~ /CET is enabled/)?"endbr32":"";
```

Comment:
I don't see how this is going to work if -fcf-protection or -fcf-protection=none is passed
to ./config.  this can only find out if the host compiler is configured to use that by default.

Reply:
I couldn't find a way to access CFLAGS from perl script.  To enable CET, I did

```
export CC="gcc -fcf-protection -Wl,-z,cet-report=error" 
```
first.

Reply:
The invocation of the generate-asm script should pass all options,
for instance here:
```
crypto/aes/aesni-sha1-x86_64.s: crypto/aes/asm/aesni-sha1-x86_64.pl
        CC="$(CC)" $(PERL) crypto/aes/asm/aesni-sha1-x86_64.pl "$(PERLASM_SCHEME)" -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include -Icrypto/include $(LIB_CFLAGS) $(LIB_CPPFLAGS) -DAES_ASM -DBSAES_ASM -DCMLL_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DX25519_ASM $(PROCESSSOR) $@
```
you should use them for such decisions (but note that PROCESSOR can be 386 or empty,
and PERLASM_SCHEME= elf or something else.

Reply:
I updated my patch with

```
# Collect all compiler flags.
my $cflags="";
for($i=0;$i<=$#ARGV;$i++) {
    if ($ARGV[$i] eq "386") {
	last;
    }
    $cflags.=" $ARGV[$i]";
}

# Check if Intel CET is enabled at compile time.
my $cet_enabled=(`echo "#ifdef __CET__\n#error CET is enabled\n#endif" | $ENV{CC} $cflags -c -o /dev/null -x c - 2>&1` =~ /CET is enabled/)?1:0;
```